components/centraldashboard/public/components/logout-button.js

```diff
@@ -0,0 +1,77 @@
+import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
+
+import '@polymer/iron-ajax/iron-ajax.js';
+import '@polymer/paper-button/paper-button.js';
+
+/**
+ * Logout button component.
+ * Handles the logout requests and post-logout redirects.
+ *
+ */
+
+export class LogoutButton extends PolymerElement {
+    static get template() {
+        return html`
+            <paper-button id="logout-button" on-tap="logout">
+                <iron-icon icon='kubeflow:logout' title="Logout"
+                </iron-icon>
+            </paper-button>
+            <iron-ajax
+                    id='logout'
+                    url='/logout'
+                    method='post'
+                    handle-as='json'
+                    headers='{{headers}}'
+                    on-response='_postLogout'>
+            </iron-ajax>
+        `;
+    }
+
+    static get properties() {
+        return {
+            headers: {
+                type: Object,
+                computed: '_setHeaders()',
+            },
+        };
+    }
+
+    /**
+     * After successful logout, redirects user to `afterLogoutURL`,
+     * received from the backend.
+     *
+     * @param {{Event}} event
+     * @private
+     */
+    _postLogout(event) {
+        window.location.replace(event.detail.response['afterLogoutURL']);
+    }
+
+    /**
+     * Call logout endpoint.
+     */
+    logout() {
+        // call iron-ajax
+        this.$.logout.generateRequest();
+    }
+
+    /**
+     * Set 'Authorization' header based on the existing cookie.
```

Comment:
looking at [this page](https://github.com/arrikto/oidc-authservice/blob/master/docs/logout.md) the authservice creates a session cookie.
does the logout button with the suggested change delete this cookie?
meaning... once logged out from one user, can we immediately log in with a new user with no issues?

Reply:
yes, a session cookie is removed after the user logs out, it is done via `Set-Cookie` header: https://github.com/arrikto/oidc-authservice/blob/ae92e8656c5252eddb305f6b9bb72ae1e1e61f6c/session.go#L75